libraries/windows_cleanup.rb

```diff
@@ -0,0 +1,15 @@
+require 'chef/handler'
+require 'chef/log'
+
+module Windows
+  class Helper
+    def do_cleanup(context)
+      Chef::Log.info 'Cleanup started.'
+      resource = context.resource_collection.lookup("windows_env[DDAGENTUSER_NAME]")
+      resource.run_action(:delete) if resource
+      resource = context.resource_collection.lookup("windows_env[DDAGENTUSER_PASSWORD]")
+      resource.run_action(:delete) if resource
+      Chef::Log.info 'Cleanup finished.'
+    end
```

Comment:
I would go for compact code and keep this helper function in `recipes/_install-windows.rb` instead of having several small files, but no strong opinion in case you prefer it this way.